ldm/invoke/patchmatch.py

```diff
@@ -0,0 +1,43 @@
+'''
+This module defines a singleton object, "patchmatch" that
+wraps the actual patchmatch object. It respects the global
+"try_patchmatch" attribute, so that patchmatch loading can
+be suppressed or deferred
+'''
+from ldm.invoke.globals import Globals
+import numpy as  np
+
+class Patchmatch (object):
+    '''
+    Thin class wrapper around the patchmatch function.
+    '''
+    
+    def __init__(self):
+        self.patch_match = None
+        self.tried_load:bool = False
+        super().__init__()
+
+    def _load_patch_match(self):
+        if self.tried_load:
+            return
+        if Globals.try_patchmatch:
+            from patchmatch import patch_match as pm
+            if pm.patchmatch_available:
+                print('>> Patchmatch initialized')
+            else:
+                print('>> Patchmatch not loaded (nonfatal)')
+            self.patch_match = pm
+        else:
+            print('>> Patchmatch loading disabled')
+        self.tried_load = True
+
+    def patchmatch_available(self)->bool:
```

Comment:
All these methods might make more sense as class methods instead of instance methods. Unless we define an abstract base class for infill methods and then reuse instances of these (which we probably will when we do nodes).

Reply:
This is good and I've made the change. Until you pointed it out I didn't realize that Python had class methods, and found this a bit weird.

Reply:
Yah I'm still getting used to the terminology for things (coming from a C-style background).